### DISCUSSION TITLE
About using linkCost

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#253] [Project Q&A 4. Etc]
In linkCost, the interface number should be given to get the cost between two routers. Since both hello packet and lsu packet are sent by multicast, I'm having trouble getting the related interface number. I used getRoutingTable() with router id to get the interface number  ,but I think this approach can be troublesome. Should I consider other approaches here?

**Comment by Pusnow** [comment]
You can use IP header's source IP address and interface's IP address.
If they in the same subnet, there is a link between the router and the interface.
